### PR TITLE
#6079: Enable multiple highlighting for OL/Leaflet DrawSupport tool

### PR DESCRIPTION
**Pull request by AGMETEOR**
## Description

This PR enables mutiple highlighting of features by the DrawSupport tool

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe: Enhancement

## Issue

**What is the current behavior?**

#6079 

**What is the new behavior?**

Highlighting multiple features

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 1]
```diff
@@ -871,19 +871,41 @@ export default class DrawSupport extends React.Component {
...
+            props = assign({}, newProps, {features: newFeatures});
         } else {
-            if (newFeature && newFeature.properties && newFeature.properties.isCircle) {
+            if (allHaveCircleProperty) {
                 props = assign({}, newProps, {features: []});
             } else {
-                props = assign({}, newProps, {features: newFeature.geometry ? [{...newFeature.geometry, properties: newFeature.properties}] : []});
+                const fts = newFeatures.reduce((pre, curr) => {
+                    if (curr.geometry) {
+                        pre.push({...curr.geometry, properties: curr.properties});
```
Comment (MV88) [inline web/client/components/map/openlayers/DrawSupport.jsx:901]: in this case this .push is fine, but be aware that it mutates the pre array, and it cannot be used inside reducers for example
return [...pre, {...curr.geometry, properties: curr.properties}];

**Review by MV88 — CHANGES_REQUESTED** [review]
Hi @AGMETEOR would be possible to add some more unit tests?

The functionality seems good

**Comment by tdipisa** [comment]
@AGMETEOR is it ready for a new review?

**Review by MV88 — APPROVED** [review]
LGTM

**Pull request merged**